packages/flutter_tools/lib/src/flx.dart

```diff
@@ -36,7 +36,7 @@ Future<int> createSnapshot({
   assert(mainPath != null);
   assert(snapshotPath != null);
   assert(packages != null);
-  final String snapshotterPath = artifacts.getArtifactPath(Artifact.genSnapshot);
+  final String snapshotterPath = artifacts.getArtifactPath(Artifact.genSnapshot, null, BuildMode.debug);
```

Comment:
Ah, now I see.  This was working for a non-local engine when I tested it, but there's an assert in CachedArtifacts.getArtifactPath that would fail if the mode is non-null.  (Gradle doesn't run flutter_tools with asserts on)

Any objection to removing the assert?
Build modes can be useful for host platforms given that binary paths will change based on the build mode